File: src/lessons/welcome/bat/bool1/SumDoubleEntity.java
package lessons.welcome.bat.bool1;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class SumDoubleEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(sumDouble((Integer)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  int sumDouble(int a, int b)
  {
    /* BEGIN SOLUTION */
    if (a == b)
      return (a + b) * 2;
    return a + b;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool1/SumDoubleEntity.scala
package lessons.welcome.bat.bool1

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class SumDoubleEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(sumDouble( param(0).asInstanceOf[Int], param(1).asInstanceOf[Int]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def sumDouble(a: Integer, b: Integer): Integer = {
		/* BEGIN SOLUTION */
	  if (a==b) {
	    return (a+b)*2
	  }
	  return a+b
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
